spec/models/api/v1/student_registration_spec.rb

```diff
@@ -29,4 +29,33 @@
     its(:student) { should be_valid }
   end
 
-end      
\ No newline at end of file
+  describe "#save" do
+    context "given duplicate user names" do
+      it "creates unique logins" do
+
+        @count = User.all.size
+
+        reg1 = API::V1::StudentRegistration.new(params)
+        reg1.save
+
+        #
+        # Set the login attribute to be the login we know was created
+        # by the previous save. Note this requires the test to have
+        # some knowledge of the implementation. Should the implementation
+        # change (or even the interface e.g. attribute access) then this
+        # test might no longer be valid.
+        #
+        # See bug PT #109387108
+        #
+        reg2 = API::V1::StudentRegistration.new(params)
+        reg2.login = 'sdoe'
+        reg2.save
+
+        expect(User.all.size).to eql(@count + 2)
```

Comment:
The test case recreates this similarly to the only way I was able to recreate by modifying the original source. That is, I changed the subclass StudentRegistration's #persist! method to replace the self.login attribute value with a value known to be present in the database prior to the call. Thus the first call to super's #persist! creates the base class' @user object with the invalid (non-unique) login. And then all subsequent attempts by the subclass to replace that value with a newly generated unique value still result in the original re-used @user object attempting to persist itself with the same invalid login.